File: Exercism Codes/PHP/BinarySearch.php
<?php

declare(strict_types=1);

function find(int $needle, array $haystack): int
{
    $left = 0;
    $right = count($haystack) - 1;

    while ($left <= $right) {
        $mid = $left + intval(($right - $left) / 2);

        if ($haystack[$mid] === $needle) {
            return $mid;
        }

        if ($haystack[$mid] > $needle) {
            $right = $mid - 1;
        }

        if ($haystack[$mid] < $needle) {
            $left = $mid + 1;
        }
    }

    return -1;
}

File: Exercism Codes/C#/BinarySearch.cs
public static class BinarySearch
{
    public static int Find(int[] input, int value)
    {
        int left = 0;
        int right = input.Length - 1;

        while (left <= right)
        {
            int mid = left + (right - left) / 2;
            if (input[mid] == value)
            {
                return mid;
            }
            else if (input[mid] < value)
            {
                left = mid + 1;
            }
            else
            {
                right = mid - 1;
            }
        }

        return -1;
    }
}